src/indicator/algorithm/calculator/envelope.js

```diff
@@ -0,0 +1,36 @@
+import d3 from 'd3';
+
+export default function() {
+
+    var channel = 0.1;
+    var midValue = function(d) {
+        return d.value;
+    };
+    var envelope = function(data) {
+        data.forEach(function(s) {
```

Comment:
as per the design guidelines, calculators should output a new array of data rather than mutating the input data.